app/src/main/java/io/apicurio/registry/storage/impl/jdbc/JdbcRegistryStorage.java

```diff
@@ -0,0 +1,646 @@
+                    .update(MetaDataKeys.TYPE, artifactType.value())
+                    .persistUpdate(entityManager, artifactId, nextVersion);
+
+            return new MetaDataMapperUpdater()
+                    .update(MetaDataKeys.TYPE, artifactType.value())
+                    .persistUpdate(entityManager, artifactId, null)
+                    .update(artifact)
+                    .toArtifactMetaDataDto();
+
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public Set<String> getArtifactIds() { // TODO paging?
+        try {
+            List<String> res1 = entityManager.createQuery(
+                    "SELECT a.artifactId FROM Artifact a", String.class)
+                    .getResultList();
+
+            return new HashSet<>(res1);
+
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+
+    // =======================================================
+
+
+    @Override
+    @Transactional
+    public ArtifactMetaDataDto getArtifactMetaData(String artifactId) throws ArtifactNotFoundException, RegistryStorageException {
+        try {
+            Artifact artifact = _getArtifact(artifactId);
+
+            return new MetaDataMapperUpdater(_getMetaData(artifactId, null))
+                    .update(artifact)
+                    .toArtifactMetaDataDto();
+
+        } catch (NoResultException ex) {
+            throw new ArtifactNotFoundException(artifactId, ex);
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void updateArtifactMetaData(String artifactId, EditableArtifactMetaDataDto metaData) throws ArtifactNotFoundException, RegistryStorageException {
+        try {
+            if (!_artifactExists(artifactId))
+                throw new ArtifactNotFoundException(artifactId);
+
+            new MetaDataMapperUpdater(_getMetaData(artifactId, null))
+                    .update(metaData)
+                    .persistUpdate(entityManager, artifactId, null);
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<String> getArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException {
+        try {
+            requireNonNull(artifactId);
+
+            if (!_artifactExists(artifactId))
+                throw new ArtifactNotFoundException(artifactId);
+
+            List<Rule> res1 = entityManager.createQuery("SELECT r FROM Rule r " +
+                    "WHERE r.artifactId = :artifact_id", Rule.class)
+                    .setParameter("artifact_id", artifactId)
+                    .getResultList();
+
+            return res1.stream().map(Rule::getName).collect(Collectors.toList());
```

Comment:
Just make such query ... "select r.name" ... no need then for stream transform.

Reply:
Done, thanks